### PR TITLE
Support multiple axes for `ht.percentile`

### PR DESCRIPTION
**Pull request by ClaudiaComito**
## Due Diligence

- General:
    - [x]  **title** of the PR is suitable to appear in the [Release Notes](https://github.com/helmholtz-analytics/heat/releases/latest)
- Implementation:
    - [x] unit tests: all split configurations tested
    - [x] unit tests: multiple dtypes tested
    - [x] documentation updated where needed

## Description

~~No changes in functionality here,~~ Tuple axes are now supported. The function is just as slow as lamented in #1389 , but it has been refactored for better readability. 

Issue/s resolved: None
Related issue: #1420 

## Changes proposed:

- refactor and use indexing instead of Bcast
- support tuple axes in `ht.percentile`
- support tuple axes in `ht.expand_dims`
**ADDED SEPT 4 2024**:
-  (following discussion with @JuanPedroGHM ) if `reshape` of a distributed array involves reducing or expanding dimensions, the output (reshaped array) will be distributed along axis 0 by default. Users can always set the `new_axis` argument to a different split axis.  

## Type of change

New feature (non-breaking change which adds functionality) plus refactoring
#### Does this change modify the behaviour of other functions? If so, which?
yes, `reshape`, see above

### CODE REVIEW

**Comment by ClaudiaComito** [comment]
@mrfh92 it might make more sense to merge #1420 first

**Comment by mrfh92** [comment]
on wait as #1420 will be merged first

**Comment by mrfh92** [comment]
#1420 has been merged meanwhile

**Comment by ClaudiaComito** [comment]
@mrfh92 I merged the changes

**Comment by ClaudiaComito** [comment]
> @ClaudiaComito so is it ready for review now?

@mrfh92 yes

File: heat/core/statistics.py [hunk 1]
```diff
@@ -1439,7 +1439,7 @@ def percentile(
 ) -> DNDarray:
     r"""
     Compute the q-th percentile of the data along the specified axis.
```
Comment (mrfh92) [inline heat/core/statistics.py:1441]:
```suggestion
    Compute the q-th percentile of the data along the specified axis.
```
Compute the q-th percentile of the data along the specified axis/axes.

File: heat/core/statistics.py [hunk 2]
```diff
@@ -1450,13 +1450,10 @@ def percentile(
         Input tensor
     q : DNDarray, scalar, or list of scalars
         Percentile or sequence of percentiles to compute. Must belong to the interval [0, 100].
-
-    axis : int, or None, optional
-        Axis along which the percentiles are computed. Default is None.
-
+    axis : int, tuple of ints, or None, optional
+        Axis along which the percentiles are computed. Default is None, corresponds to calculating the percentile over the flattened array.
```
Comment (mrfh92) [inline heat/core/statistics.py:1454]:
```suggestion
        Axis (if int) or axes (if tuple) along which the percentiles are computed. Default is None, corresponds to calculating the percentile over the flattened array.
```

File: heat/core/tests/test_statistics.py [hunk 3]
```diff
@@ -1157,14 +1157,15 @@ def test_percentile(self):
         q = 15.9
         for dim in range(x_ht.ndim):
             p_np = np.percentile(x_np, q, axis=dim)
+            p_np_keepdims = np.percentile(x_np, q, axis=dim, keepdims=True)
```
Comment (mrfh92) [inline heat/core/tests/test_statistics.py:1160]: The combination of sketched percentile with the new option to take percentile along a tuple of axes is not testes. Would that work or should we just throw an error?

**Review by mrfh92 — CHANGES_REQUESTED** [review]
This looks fine; I would only suggest two changes in the docs and moreover, I am not sure what happens when you combine sketched percentile with the option to allow for a tuple of axes to take percentile along.

File: heat/core/tests/test_statistics.py (hunk 3 above)
Comment (ClaudiaComito) [reply heat/core/tests/test_statistics.py:1160]: Great you noticed, I implemented the tests and found an error in `reshape` (fixed).

**Comment by ClaudiaComito** [comment]
I found a problem while writing tests for the last uncovered lines of code. Will fix it when I'm back.

**Comment by ClaudiaComito** [comment]
@mrfh92 @JuanPedroGHM I think I'm done here.

@mrfh92 I expanded the tests to cover tuple axes for sketched percentile, and also writing to `out` buffer. @JuanPedroGHM and I also decided to make some changes in `reshape`, see expanded description above. Let me know if something's missing, thanks!

**Review by mrfh92 — DISMISSED** [review]
My change requests have been addressed. 
Thanks for the work 👍

**Comment by ClaudiaComito** [comment]
@mrfh92 I made small edits to the docs to trigger the pipeline again, codecov had gotten stuck. I'm afraid you need to approve again, thanks!

**Review by mrfh92 — APPROVED** [review]

**Pull request merged**